rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/ExportNodeSourceToCatalogServlet.java

```diff
@@ -0,0 +1,116 @@
+import org.ow2.proactive_grid_cloud_portal.rm.server.ServletRequestTransformer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Servlet invoked when a node source is exported to the catalog. The request
+ * content is expected to be in JSON format, containing the configuration of a
+ * node source.
+ */
+public class ExportNodeSourceToCatalogServlet extends HttpServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportNodeSourceToCatalogServlet.class);
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        LOGGER.info("Calling POST");
+        upload(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        upload(request, response);
+    }
+
+    private void upload(HttpServletRequest request, HttpServletResponse response) {
+        response.setContentType("text/html");
+        File nodeSourceJsonFile = null;
+        try {
+            String sessionId = null;
+            CatalogObject catalogObject = new CatalogObject();
+            for (FileItem formItem : new ServletRequestTransformer().getFormItems(request)) {
+                if (formItem.isFormField()) {
+                    if (formItem.getFieldName().equals("sessionId")) {
+                        sessionId = formItem.getString();
+                    } else if (formItem.getFieldName().equals(BUCKET_NAME_PARAM)) {
+                        catalogObject.setBucketName(formItem.getString());
+                    } else if (formItem.getFieldName().equals(NAME_PARAM)) {
+                        catalogObject.setNodeSourceName(formItem.getString());
+                    } else if (formItem.getFieldName().equals(FILE_CONTENT_PARAM)) {
+                        nodeSourceJsonFile = File.createTempFile("node-source-configuration", ".json");
+                        new PrintWriter(nodeSourceJsonFile).write(formItem.getString());
+                        formItem.write(nodeSourceJsonFile);
+                        catalogObject.setNodeSourceJsonFile(nodeSourceJsonFile);
+                    } else if (formItem.getFieldName().equals(KIND_PARAM)) {
+                        catalogObject.setKind(formItem.getString());
+                    } else if (formItem.getFieldName().equals(COMMIT_MESSAGE_PARAM)) {
+                        catalogObject.setCommitMessage(formItem.getString());
+                    } else if (formItem.getFieldName().equals(OBJECT_CONTENT_TYPE_PARAM)) {
+                        catalogObject.setObjectContentType(formItem.getString());
+                    }
+                }
+                formItem.delete();
+            }
+            CatalogRequestBuilder catalogRequestBuilder = new CatalogRequestBuilder(catalogObject);
+            String requestUri = catalogRequestBuilder.build();
+            LOGGER.info("Node Source exported to catalog using resource URI: " + requestUri);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)